src/extensions/extensionInterface.ts

```diff
@@ -269,6 +295,45 @@ export const recordingSelectionReducer: Reducer<RecordingSelection, RecordingSel
             ampScaleFactor: (state.ampScaleFactor || 1) * action.multiplier
         }
     }
+    else if (action.type === 'SetCurrentTimepointVelocity') {
+        return {
+            ...state,
+            animation: {
+                ...(state.animation || {}),
+                lastUpdateTimestamp: Number(new Date()),
+                currentTimepointVelocity: action.velocity
+            }
+        }
+    }
+    else if (action.type === 'AnimateRecording') {
+        const lastUpdate = state?.animation?.lastUpdateTimestamp || Number(new Date())
+        const current = Number(new Date())
+        const elapsed = current - lastUpdate
+        if (elapsed !== 0) {
+            let state2 = {...state}
+            const currentTimepointVelocity = state?.animation?.currentTimepointVelocity || 0
+            const currentTimepoint = state.currentTimepoint
+            let somethingChanged = false
+            if ((currentTimepointVelocity) && (currentTimepoint !== undefined)) {
```

Comment:
This is a little confusing--`currentTimepointVelocity` has a default zero value. So is this part of the check making sure that the velocity is nonzero, i.e. we're going to advance the animation? If that's the intent, it might be clearer to do the zero comparison explicitly.